pkg/cmd/pulumi/package_publish.go

```diff
@@ -0,0 +1,175 @@
+			ctx := commandContext()
+			return publCmd.Run(ctx, args)
+		}),
+	}
+	cmd.PersistentFlags().StringVar(&publCmd.Path, "path", "",
+		`The path to the root of your package.
+	Example: ./sdk/nodejs
+	`)
+	return cmd
+}
+
+type publishCmd struct {
+	Path string
+}
+
+func (cmd *publishCmd) Run(ctx context.Context, args []string) error {
+	pkg := args[0]
+	lang := args[1]
+
+	logging.V(1).Infof("publishing package %s to the %s registry \n", pkg, lang)
+	switch lang {
+	case "nodejs":
+		err := publishToNPM(pkg, cmd.Path)
+		if err != nil {
+			return err
+		}
+	case "all", "python", "java", "dotnet":
+		return fmt.Errorf("support for %q coming soon", lang)
+	default:
+		return fmt.Errorf("unsupported language %q", lang)
+	}
+
+	return nil
+}
+
+func publishToNPM(pkg, path string) error {
+	// verify path
+	_, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("reading file %s: %w", path, err)
+	}
+
+	// Verify npm exists and is set up: npm, user login
+	npm, err := executable.FindExecutable("npm")
+	if err != nil {
+		return err
+	}
+
+	// verify auth for npm
+	whoamiCmd := exec.Command(npm, "whoami")
+	whoamiCmd.Stderr = os.Stderr
+	whoami, err := whoamiCmd.Output()
+	if err != nil {
+		return err
+	}
+
+	logging.V(1).Infof("Logged in as %s", whoami)
+
+	// TODO: possibly check package dependencies
+
+	var pkgInfo struct {
+		Name    string `json:"name"`
+		Version string `json:"version"`
+	}
+
+	file, err := os.ReadFile(filepath.Join(path, "package.json"))
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(file, &pkgInfo)
+	if err != nil {
+		return fmt.Errorf("unmarshal package.json: %w", err)
+	}
+
+	// Determine which tag to set
+	// npm adds `latest` as the default tag, and we want that to mean the newest released version.
+	var npmTag string
+
+	switch {
+	case strings.Contains(pkgInfo.Version, "-alpha"):
+		npmTag = "dev"
+	case strings.Contains(pkgInfo.Version, "dev"):
+		npmTag = "dev"
+	case strings.Contains(pkgInfo.Version, "-beta"):
+		npmTag = "beta"
+	case strings.Contains(pkgInfo.Version, "-rc"):
+		npmTag = "rc"
+	default:
+		npmTag = "latest"
+	}
+
+	pkgNameWithVersion := pkgInfo.Name + "@" + pkgInfo.Version
+
+	// Verify version doesn't already exist
+	infoCmd := exec.Command(npm, "info", pkgNameWithVersion)
+	infoCmd.Stderr = os.Stderr
+	logging.V(1).Infof("Running %s\n", infoCmd)
+	output, err := infoCmd.Output()
+	if err != nil {
+		return err
+	}
+
+	if len(output) > 0 {
+		return fmt.Errorf("publish %s failed, likely because the version already exists", pkg)
+	}
+
+	logging.V(1).Infof("The version does not exist yet, and it is safe to publish")
+	fmt.Printf("Publishing %s to npm package registry...\n", pkg)
+	npmPublishCmd := exec.Command(npm, "publish", path, "-tag", npmTag)
+	// Pass through command output and error
+	npmPublishCmd.Stdout = os.Stdout
+	npmPublishCmd.Stderr = os.Stderr
+	err = npmPublishCmd.Run()
+	// handle errors
+	if err != nil {
```

Comment:
minor nit: I would omit these comments.

```suggestion
	npmPublishCmd.Stdout = os.Stdout
	npmPublishCmd.Stderr = os.Stderr
	if err := npmPublishCmd.Run(); err != nil {
```